src/main/java/io/axoniq/axonserver/connector/admin/impl/AdminChannelImpl.java

```diff
@@ -91,6 +127,170 @@ private EventProcessorIdentifier eventProcessorId(String eventProcessorName, Str
+        FutureStreamObserver<Token> responseObserver = new FutureStreamObserver<>(null);
+        applicationServiceStub.createOrUpdateApplication(request, responseObserver);
+        return responseObserver;
+    }
+
+    @Override
+    public CompletableFuture<ApplicationOverview> getApplication(String applicationName)  {
+        FutureStreamObserver<ApplicationOverview> responseObserver = new FutureStreamObserver<>(null);
+        applicationServiceStub.getApplication(ApplicationId.newBuilder().setApplicationName(applicationName).build(), responseObserver);
+        return responseObserver;
+    }
+
+    @Override
+    public CompletableFuture<Token> refreshToken(String applicationName)  {
+        FutureStreamObserver<Token> responseObserver = new FutureStreamObserver<>(null);
+        applicationServiceStub.refreshToken(ApplicationId.newBuilder().setApplicationName(applicationName).build(), responseObserver);
+        return responseObserver;
+    }
+
+    @Override
+    public CompletableFuture<Void> deleteApplication(String applicationName)  {
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
+        applicationServiceStub.deleteApplication(ApplicationId.newBuilder().setApplicationName(applicationName).build(), responseObserver);
+        return responseObserver.thenAccept(empty -> {
+        });
+    }
+
+    @Override
+    public CompletableFuture<Void> createContext(CreateContextRequest request)  {
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
+        contextServiceStub.createContext(request, responseObserver);
+        return responseObserver.thenAccept(empty -> {
+        });
+    }
+
+    @Override
+    public CompletableFuture<Void> deleteContext(DeleteContextRequest request)  {
+        FutureStreamObserver<Empty> responseObserver = new FutureStreamObserver<>(null);
+        contextServiceStub.deleteContext(request, responseObserver);
+        return responseObserver.thenAccept(empty -> {
+        });
+    }
+
+    @Override
+    public CompletableFuture<ContextOverview> getContextOverview(String context)  {
+        FutureStreamObserver<ContextOverview> responseObserver = new FutureStreamObserver<>(null);
+        contextServiceStub.getContext(GetContextRequest.newBuilder().setName(context).build(), responseObserver);
+        return responseObserver;
+    }
+
+    @Override
+    public CompletableFuture<List<ContextOverview>> getAllContexts() {
+        FutureListStreamObserver<ContextOverview> responseObserver = new FutureListStreamObserver<>();
+        contextServiceStub.getContexts(Empty.newBuilder().build(),responseObserver);
+        return responseObserver;
+    }
+
+
+    @Override
+    public ResultStream<ContextUpdate> subscribeContextUpdates() {
+        AbstractBufferedStream<ContextUpdate, Empty> results = new AbstractBufferedStream<ContextUpdate, Empty>(
+                "", 32, 8
```

Comment:
What are these magic numbers for? 
I think it's good if we either give named parameters to make this clear or to promote this `AbstractBufferedStream` to an inner private class for clarity.

Reply:
There is same usage in `QueryChannelImpl.java,` inside `ResultStream<QueryResponse> query(QueryRequest query)`.  So I have just reused it... Magic numbers are same as in query, and they represent buffer size and refill batch.

Reply:
I must say, that also in the `QueryChannelImpl` I don't think it's nice to look at "random" numbers instead of named parameters. Hence, I personally don't think that's an argument to note use named fields/constants for this :sweat_smile: